pkg/quarterdeck/middleware/auth.go

```diff
@@ -128,6 +130,24 @@ func GetClaims(c *gin.Context) (*tokens.Claims, error) {
 	return claims.(*tokens.Claims), nil
 }
 
+// ContextFromRequest creates a context from the gin request context, copying fields
+// that may be required for forwarded requests. This method should be called by
+// handlers which need to forward requests to other services and need to preserve data
+// from the original request such as the user's credentials.
+func ContextFromRequest(c *gin.Context) (ctx context.Context, err error) {
```

Comment:
Note that I didn't end up creating an interface implementation of the passthrough credentials and used the `Token` one instead, let me know what you think.

Reply:
I'm willing to give this a shot to see how we like using it!